neo4jgraph.py

```diff
@@ -0,0 +1,457 @@
+            Dataset collection this dataset belongs to.
+
+        slo: str
+            Represents how often the dataset is written.
+
+        Raises
+        ------
+        ServiceUnavailable
+            Web socket error that occurs when having problems connecting to neo4j.
+        """
+        with self.driver.session() as session:
+            try:
+                session_log = session.write_transaction(self._generate_dataset, dataset_id, dataset_collection_id, slo)
+                for row in session_log:
+                    logging.info(f'Generated dataset {row["dataset"][0]} with regex grouping {row["dataset"][1]} '
+                                 f'and dataset collection {row["dataset_collection"]}')
+            except ServiceUnavailable as exception:
+                logging.error(exception)
+                raise
+
+    @staticmethod
+    def _generate_dataset(tx, dataset_id, dataset_collection_id, slo):
+        """Creates and runs a query for dataset creation. Creates a link to a corresponding dataset collection."""
+        query = (
+            f'MATCH (dataset_collection:dataset_collection) '
+            f'WHERE dataset_collection.dataset_collection_id = {dataset_collection_id} '
+            f'MERGE (dataset:dataset {{dataset_id: {dataset_id}, dataset_collection_id: {dataset_collection_id}, '
+            f'regex_grouping: "data.{dataset_id}.*", dataset_name: "dataset.{dataset_id}", '
+            f'dataset_description: "Dataset number {dataset_id}", slo: "{slo}" }}) '
+            f'MERGE (dataset_collection) -[:CONTAINS] -> (dataset) '
+            f'RETURN dataset_collection, dataset'
+        )
+
+        query_output = tx.run(query)
+        logging.info(f"Run cypher query to create a dataset: {query}.")
+        return [{"dataset_collection": row["dataset_collection"]["dataset_collection_id"],
+                 "dataset": (row["dataset"]["dataset_id"], row["dataset"]["regex_grouping"])} for row in query_output]
+
+    def generate_system(self, system_id, system_critic, system_collection_id):
+        """Generates a system with id and description based on the id.
+        Creating a system with id 1, collection id 2, system_critic 'LEVEL_1' will create a node with these attributes:
+            system_id: 1,
+            system_collection_id: 2
+            regex_grouping: system.1.*
+            system_name: system.1
+            system_description: System number 1
+            system_critic: LEVEL_1
+
+        Creating a connection between system with id 1 and system collection with id 2.
+
+        Parameters
+        ----------
+        system_id : int
+            System id, is unique.
+
+        system_critic: str
+            System criticality type.
+
+        system_collection_id: int
+            System collection this system belongs to.
+
+        Raises
+        ------
+        ServiceUnavailable
+            Web socket error that occurs when having problems connecting to neo4j.
+        """
```

Comment:
nit: format the docstring similarly to https://google.github.io/styleguide/pyguide.html#383-functions-and-methods, eg:

Args:
  arg1:

instead of 
Parameters
_______
  arg1: